pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -14,6 +14,11 @@ func (d *DataplaneResource) Validate() error {
 	return err.OrNil()
 }
 
+// For networking section validation we need to take into account our legacy model.
+// Legacy model is detected by having interface defined either on inbound listeners.
+// We do not allow networking.address with the old format. Instead, we recommend switching to the new format.
+// When we've got dataplane in the new format, we require networking.address field to be defined unless it is Gateway
```

Comment:
> unless it is Gateway

What is the value in maintaining backwards compatibility if it leads to a situation where it's not possible to scrape metrics out of a `Dataplane` ?

Don't we want to prevent it from happening in the first place ?

Reply:
Ok, we agreed on the call that we should require the address.